pkg/streamingpromql/operators/functions/range_vectors.go

```diff
@@ -369,3 +369,62 @@ func avgHistograms(head, tail []promql.HPoint) (*histogram.FloatHistogram, error
 
 	return avgSoFar, nil
 }
+
+var Deriv = FunctionOverRangeVectorDefinition{
+	SeriesMetadataFunction: DropSeriesName,
+	StepFunc:               deriv,
+}
+
+func deriv(step types.RangeVectorStepData, _ float64, _ types.EmitAnnotationFunc) (float64, bool, *histogram.FloatHistogram, error) {
+	head, tail := step.Floats.UnsafePoints(step.RangeEnd)
+
+	if len(head) < 2 && len(tail) == 0 {
```

Comment:
There's no guarantee the first two points will be in the head:

```suggestion
	if (len(head) + len(tail)) < 2 {
```

Reply:
That's true! Thanks.